tests/test_utils.py

```diff
@@ -198,5 +199,5 @@ def test_cleanup(self):
     def test_resolve_dep_oe(self):
         """Test resolve dependency with Open Embedded"""
         # Note(allenh1): we're not going to test the hard-coded resolutions.
-        self.assertEqual(resolve_dep('tinyxml2', 'oe'), 'libtinyxml2')
-        self.assertEqual(resolve_dep('p2os_msgs', 'oe'), 'p2os-msgs')
```

Comment:
Yeah, I can see that. Tried querying it by rosdep resolve:

$ rosdep resolve p2os_msgs
ERROR: no rosdep rule for 'p2os_msgs'
$ rosdep --os=ubuntu:bionic resolve p2os_msgs
ERROR: no rosdep rule for 'p2os_msgs'
$ rosdep --os=openembedded:* resolve p2os_msgs
ERROR: no rosdep rule for 'p2os_msgs'

$ ./env/bin/rosdep resolve libdw-dev
#apt
libdw-dev
$ ./env/bin/rosdep --os=ubuntu:bionic resolve libdw-dev
#apt
libdw-dev
$ ./env/bin/rosdep --os=openembedded:* resolve libdw-dev
#opkg
elfutils@openembedded-core

And then when running the standalone tests:
ERROR: Test resolve dependency with Open Embedded
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andre/projects/superflore/tests/test_utils.py", line 203, in test_resolve_dep_oe
    self.assertEqual(resolve_dep('p2os_msgs', 'openembedded')[0], ['p2os_msgs'])
  File "/home/andre/projects/superflore/superflore/utils.py", line 209, in resolve_dep
    return resolve_rosdep_key(pkg, 'openembedded', '', distro)
  File "/home/andre/projects/superflore/superflore/rosdep_support.py", line 101, in resolve_rosdep_key
    .format(key, os_name)
superflore.exceptions.UnresolvedDependency: could not resolve package p2os_msgs for os openembedded

Do you think resolve_dep should catch UnresolvedDependency internally and output the potential OpenEmbedded recipe name for it or you mean something else?

Reply:
So I don't expect that to resolve with rosdep -- since it's a ROS package, it won't resolve with rosdep (at least to my knowledge).

Reply:
Yes, exactly. It'll handle the external dependencies.

Reply:
> Yes, exactly. It'll handle the external dependencies.

My mistake -- you're right. I forgot why I added that test case... This should be expected to fail, since the behavior is now corrected.

Reply:
Thanks, I've updated the expectation in [the test](https://github.com/ros-infrastructure/superflore/pull/168/files#diff-87010571bbeeee0b9a13279c493670f3R200)